dev/session_metrics/lib/slurm_metrics/metrics_session_service.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module SlurmMetrics
+  # Service to manages retrieving and storing the user job metrics.
+  class MetricsSessionService
+    attr_reader :cluster
+
+    def initialize
+      @cluster = Configuration.job_clusters.select(&:slurm?).first
+    end
+
+    def read_job_metrics(session)
+      file_path = job_metrics_path(session.id)
+      return refresh_job_metrics(session) unless file_path.exist?
+
+      # DEFAULT VALUE
+      yml = {}
+      begin
+        yml = YAML.safe_load(file_path.read).deep_symbolize_keys || {}
+        SlurmMetrics::SessionMetrics.new(yml)
+      rescue => e
+        Rails.logger.error("Can't read or parse job metrics: #{file_path} because of error #{e}")
+      end
+
+      SlurmMetrics::SessionMetrics.new(yml)
```

Comment:
If I compare the previous code snippet with this one I found something weird regarding `SlurmMetrics::SessionMetrics.new(yml)` . Line 20 of this file seems weird to me if we are doing the same in line 25. We are not saving the value in line 20. I guess that refactoring the code to this could make more sense:

```ruby
      # DEFAULT VALUE
      session_metrics = SlurmMetrics::SessionMetrics.new
      begin
        yml = YAML.safe_load(file_path.read).deep_symbolize_keys || {}
        session_metrics = SlurmMetrics::SessionMetrics.new(yml)
      rescue => e
        Rails.logger.error("Can't read or parse job metrics: #{file_path} because of error #{e}")
      end

      session_metrics
```

Reply:
ahh - I made a change before and forgot to delete line 20. Line 20 should be deleted. Will fix.